mayastor/src/grpc/mayastor/v1.rs

```diff
@@ -0,0 +1,196 @@
+        let r = fut.await;
+
+        // Request completed, remove the marker.
+        let ctx = guard.take().expect("gRPC context disappeared");
+
+        match r {
+            Ok(r) => r,
+            Err(_e) => {
+                warn!("{}: gRPC method panicked, args: {}", ctx.id, ctx.args);
+                Err(Status::cancelled(format!(
+                    "{}: gRPC method panicked",
+                    ctx.id
+                )))
+            }
+        }
+    }
+}
+
+impl TryFrom<CreatePoolRequest> for PoolArgs {
+    type Error = LvsError;
+    fn try_from(pool: CreatePoolRequest) -> Result<Self, Self::Error> {
+        if pool.disks.is_empty() {
+            Err(LvsError::Invalid {
+                source: Errno::EINVAL,
+                msg: "Missing devices".to_string(),
+            })
+        } else {
+            Ok(Self {
+                name: pool.name,
+                disks: pool.disks,
+            })
+        }
+    }
+}
+
+impl MayastorSvc {
+    pub fn new(interval: Duration) -> Self {
+        Self {
+            name: String::from("CSISvc"),
+            interval,
+            rw_lock: tokio::sync::RwLock::new(None),
+        }
+    }
+}
+
+impl From<Lvs> for Pool {
+    fn from(l: Lvs) -> Self {
+        Self {
+            name: l.name().into(),
+            disks: vec![l.base_bdev().bdev_uri().unwrap_or_else(|| "".into())],
+            state: PoolState::PoolOnline.into(),
+            capacity: l.capacity(),
+            used: l.used(),
+        }
+    }
+}
+
+#[tonic::async_trait]
+impl mayastor_server::Mayastor for MayastorSvc {
+    #[named]
+    async fn create_pool(
+        &self,
+        request: Request<CreatePoolRequest>,
+    ) -> GrpcResult<Pool> {
+        self.locked(
+            GrpcClientContext::new(&request, function_name!()),
+            async move {
+                let args = request.into_inner();
+
+                if args.disks.is_empty() {
+                    return Err(Status::invalid_argument("Missing devices"));
+                }
+
+                let rx = rpc_submit::<_, _, LvsError>(async move {
+                    let pool = Lvs::create_or_import(PoolArgs::try_from(args)?)
+                        .await?;
+                    // Capture current pool config and export to file.
+                    PoolConfig::capture().export().await;
+                    Ok(Pool::from(pool))
+                })?;
+
+                rx.await
+                    .map_err(|_| Status::cancelled("cancelled"))?
+                    .map_err(Status::from)
+                    .map(Response::new)
+            },
+        )
+        .await
+    }
+
+    #[named]
+    async fn destroy_pool(
+        &self,
+        request: Request<DestroyPoolRequest>,
+    ) -> GrpcResult<Null> {
+        self.locked(
+            GrpcClientContext::new(&request, function_name!()),
+            async move {
+                let args = request.into_inner();
+                info!("{:?}", args);
```

Comment:
Shall we  make it debug() ?

Reply:
IMO info makes more sense as user should get an history of the volumes which got deleted. The details of deletion process can be in debug.